MovieEnthusiast.Api/Controllers/ApiControllerBase.cs

```diff
@@ -0,0 +1,13 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace MovieEnthusiast.Api.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
+public class ApiControllerBase : ControllerBase
+{
+    private ISender? _mediator;
```

Comment:
I think it is neat to have `ISender` available for derived controllers. However, MS does not recommend this way of resolving dependencies [ref](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-7.0#request-services).
I suggest just removing this for now and resolving this dependency in derived controllers via constructor